build.sh

```
@@ -2,10 +2,127 @@
+        echo "Linux detected, ensuring required packages are installed..."
+        missing_packages=()
+        for pkg in "${REQUIRED_PACKAGES[@]}"; do
+            if ! dpkg -s "$pkg" &> /dev/null; then
+                missing_packages+=("$pkg")
+            fi
+        done
+        
+        if [ ${#missing_packages[@]} -gt 0 ]; then
+            if [[ $EUID -ne 0 ]]; then
+                echo "Installing missing dependencies requires root privileges."
+                echo "Please re-run with: sudo ./build.sh --install-deps $@"
```

Shellcheck (0.11.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)